searx/engines/moviepilot.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Moviepilot
+
+Moviepilot is a German movie database, similar to IMDB or TMDB.
+It doesn't have any official API, but it uses JSON requests internally
+to fetch search results and suggestions, that's being used in this
+implementation.
+
+Moviepilot additionally allows to discover movies by certain categories
+or filters, hence we provide the following syntax:
+- Any normal search query -> Fetch search results by the query
+- A query containing one of the category identifiers `fsk`, `genre`, `jahr`,
+`jahrzent`, `land`, `online`, `stimmung` will be used to search trending items
+by the specified `discovery_type` and the provided filters, which are appended
+to the filter category after a `-`.
+
+Search examples:
+- Normal: !mp Tom Cruise
+- By filter: !mp fsk-0 land-deutschland genre-actionfilm
+- By filter: !mp jahrzehnt-2020er online-netflix
+
+For a list of all public filters, observe the url path when browsing
+https://www.moviepilot.de/filme/beste.
+"""
+
+from urllib.parse import urlencode
+from searx.utils import html_to_text
+
+about = {
+    'website': "https://www.moviepilot.de",
+    'official_api_documentation': None,
+    'use_official_api': False,
+    'require_api_key': False,
+    'results': 'JSON',
+    'language': 'de',
+}
+paging = True
+
+base_url = "https://www.moviepilot.de"
+image_url = "https://assets.cdn.moviepilot.de/files/{image_id}/fill/155/223/{filename}"
+
+discovery_type = "filme"  # any of filme, serien, person
```

Comment:
>  any of filme, serien, person

is `person` really a discovery-type?  As far I can see only ...

![image](https://github.com/searxng/searxng/assets/554536/d44d741c-9e5d-42b0-8d64-2493610c53f0)

are discovery types and `person` is a filter .. or I'm wrong?

Reply:
Oh yes, that's right!

I just saw the `type` set to "person" in some search results, so I thought it would be supported too, but it doesn't really make sense to filter by genre or FSK of a person :joy:

Reply:
It looks like we can completely drop the `type` query in the discovery and search for series and movies at the same time, which makes much more sense!

Reply:
I thought it would be a required parameter, but I just tested it: If we don't specify the `type`, we get series and movies.